pkg/cloudprovider/types.go

```diff
@@ -49,6 +53,27 @@ type NodeGroup interface {
 	Stabilized() (bool, string, error)
 }
 
+// Fleet represents a fleet request in the cloud provider,
+// all the instances in the fleet will have the same properties
+// number of instances can be controlled by setting the capacity
+type Fleet interface {
+	// SetAvailabilityZone is the zone the instances will run in this fleet
+	SetAvailabilityZone(zone string)
```

Comment:
Instead of setters, I'd just define a struct (maybe NodeConstraints or just Constraints?) and pass that to nodeClient.Create(ctx, { ... })

Reply:
I just wanted to make it cleaner for the client to be able to set the parameters, I was trying to be consistent with NodeGroup interface and abstract this provider specific stuff from client.

If I define a struct I would have to define it in either of these two packages - `pkg/cloudprovider` or `pkg/cloudprovider/aws`, latter will make it AWS specific and if I make it in the `cloudprovider` package, I will have to import that package in `pkg/cloudprovider/aws`

Reply:
Discussed offline. These details should live in the AWS cloud provider.